Can't build hlint or stylish-haskell

I have almost everything building except hlint or stylish-haskell:

```
[nix-shell:~/work/clc-stackage]$ cabal build -w /home/ollie/work/ghc/_build/stage1/bin/ghc --keep-going -j10
Build profile: -w ghc-9.2.5 -O1
In order, the following will be built (use -v for more details):
 - hlint-3.4.1 (lib) (requires build)
 - stylish-haskell-0.14.2.0 (lib) (requires build)
 - clc-stackage-0.1.0.0 (lib) (first run)
Starting     stylish-haskell-0.14.2.0 (lib)
Starting     hlint-3.4.1 (lib)
Building     stylish-haskell-0.14.2.0 (lib)
Building     hlint-3.4.1 (lib)

Failed to build hlint-3.4.1.
Build log (
/home/ollie/.cabal/logs/ghc-9.2.5/hlint-3.4.1-8edf210b78466f00a48344cab77a270c537cc33da356fd984fd92e3df63dbf8b.log
):
Configuring library for hlint-3.4.1..
Preprocessing library for hlint-3.4.1..
Building library for hlint-3.4.1..
[ 1 of 60] Compiling EmbedData        ( src/EmbedData.hs, dist/build/EmbedData.o, dist/build/EmbedData.dyn_o )

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libgmp.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libtinfo.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).
[ 2 of 60] Compiling Extension        ( src/Extension.hs, dist/build/Extension.o, dist/build/Extension.dyn_o )

src/Extension.hs:67:25: error:
    • Couldn't match type ‘ghc-lib-parser-9.2.3.20220527:GHC.LanguageExtensions.Type.Extension’
                     with ‘Extension’
      Expected: [(Extension, ([Extension], [Extension]))]
        Actual: [(ghc-lib-parser-9.2.3.20220527:GHC.LanguageExtensions.Type.Extension,
                  ([ghc-lib-parser-9.2.3.20220527:GHC.LanguageExtensions.Type.Extension],
                   [ghc-lib-parser-9.2.3.20220527:GHC.LanguageExtensions.Type.Extension]))]
      NB: ‘Extension’
            is defined in ‘GHC.LanguageExtensions.Type’
                in package ‘ghc-boot-th-9.2.5’
          ‘ghc-lib-parser-9.2.3.20220527:GHC.LanguageExtensions.Type.Extension’
            is defined in ‘GHC.LanguageExtensions.Type’
                in package ‘ghc-lib-parser-9.2.3.20220527’
    • In the expression: GhclibParserEx.extensionImplications
      In an equation for ‘extensionImplications’:
          extensionImplications = GhclibParserEx.extensionImplications
   |
67 | extensionImplications = GhclibParserEx.extensionImplications
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Failed to build stylish-haskell-0.14.2.0.
Build log (
/home/ollie/.cabal/logs/ghc-9.2.5/stylish-haskell-0.14.2.0-0f30f0ac2bf98e418f720288b325988d67e4516d4f08a0e4885ab735003b86f6.log
):
Configuring library for stylish-haskell-0.14.2.0..
Preprocessing library for stylish-haskell-0.14.2.0..
Building library for stylish-haskell-0.14.2.0..
[ 1 of 26] Compiling Language.Haskell.Stylish.Block ( lib/Language/Haskell/Stylish/Block.hs, dist/build/Language/Haskell/Stylish/Block.o, dist/build/Language/Haskell/Stylish/Block.dyn_o )
[ 2 of 26] Compiling Language.Haskell.Stylish.Config.Internal ( lib/Language/Haskell/Stylish/Config/Internal.hs, dist/build/Language/Haskell/Stylish/Config/Internal.o, dist/build/Language/Haskell/Stylish/Config/Internal.dyn_o )
[ 3 of 26] Compiling Language.Haskell.Stylish.Editor ( lib/Language/Haskell/Stylish/Editor.hs, dist/build/Language/Haskell/Stylish/Editor.o, dist/build/Language/Haskell/Stylish/Editor.dyn_o )
[ 4 of 26] Compiling Language.Haskell.Stylish.GHC ( lib/Language/Haskell/Stylish/GHC.hs, dist/build/Language/Haskell/Stylish/GHC.o, dist/build/Language/Haskell/Stylish/GHC.dyn_o )

lib/Language/Haskell/Stylish/GHC.hs:71:32: error:
    • Couldn't match expected type ‘GHC.Settings’
                  with actual type ‘ghc-lib-parser-9.2.3.20220527:GHC.Settings.Settings’
      NB: ‘GHC.Settings’
            is defined in ‘GHC.Settings’ in package ‘ghc-9.2.5’
          ‘ghc-lib-parser-9.2.3.20220527:GHC.Settings.Settings’
            is defined in ‘GHC.Settings’
                in package ‘ghc-lib-parser-9.2.3.20220527’
    • In the first argument of ‘defaultDynFlags’, namely
        ‘GHCEx.fakeSettings’
      In the expression:
        defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
      In an equation for ‘baseDynFlags’:
          baseDynFlags
            = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |
71 | baseDynFlags = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |                                ^^^^^^^^^^^^^^^^^^

lib/Language/Haskell/Stylish/GHC.hs:71:51: error:
    • Couldn't match expected type ‘GHC.LlvmConfig’
                  with actual type ‘ghc-lib-parser-9.2.3.20220527:GHC.Driver.Session.LlvmConfig’
      NB: ‘GHC.LlvmConfig’
            is defined in ‘GHC.Driver.Session’ in package ‘ghc-9.2.5’
          ‘ghc-lib-parser-9.2.3.20220527:GHC.Driver.Session.LlvmConfig’
            is defined in ‘GHC.Driver.Session’
                in package ‘ghc-lib-parser-9.2.3.20220527’
    • In the second argument of ‘defaultDynFlags’, namely
        ‘GHCEx.fakeLlvmConfig’
      In the expression:
        defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
      In an equation for ‘baseDynFlags’:
          baseDynFlags
            = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |
71 | baseDynFlags = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |                                                   ^^^^^^^^^^^^^^^^^^^^
Error: cabal: Failed to build hlint-3.4.1 (which is required by
clc-stackage-0.1.0.0). See the build log above for details.
Failed to build stylish-haskell-0.14.2.0 (which is required by
clc-stackage-0.1.0.0). See the build log above for details.
```

I know `hlint` has flags to control whether or not to use `ghc-lib-parser`, but before I do anything I wanted to know if anyone else was experiencing this.

--- a/cabal.project
+++ b/cabal.project
@@ -22,3 +22,7 @@ constraints:
 allow-newer:
   aura:bytestring,
   aura:time
+
+constraints: hlint +ghc-lib
+constraints: ghc-lib-parser-ex -auto
+constraints: stylish-haskell +ghc-lib

